File: tools/connect_ollama.py

import requests
import os
import sys
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

OLLAMA_URL = os.getenv("OLLAMA_URL", "http://localhost:11434/api/chat")
MODEL = os.getenv("MODEL", "llama3.2")

def check_ollama():
    url = OLLAMA_URL
    
    # Determine if we are using chat or generate endpoint to format payload
    if "/api/chat" in url:
        payload = {
            "model": MODEL,
            "messages": [{"role": "user", "content": "Hello via Python!"}],
            "stream": False
        }
    else:
        payload = {
            "model": MODEL,
            "prompt": "Hello via Python!",
            "stream": False
        }
    
    print(f"Connecting to Ollama at {url} with model {MODEL}...")
    
    try:
        response = requests.post(url, json=payload)
        response.raise_for_status()
        
        data = response.json()
        print("✅ Connection Successful!")
        
        if "/api/chat" in url:
            content = data.get("message", {}).get("content", "No content found")
            print(f"Response: {content}")
        else:
            print(f"Response: {data.get('response', 'No response field')}")
        
    except requests.exceptions.RequestException as e:
        print(f"❌ Connection Failed: {e}")
        sys.exit(1)

if __name__ == "__main__":
    check_ollama()

File: tools/connect_ollama.js

// No imports needed for Node 18+ (fetch is global)
// If on older node, we might fail, but let's assume recent environment.

async function checkOllama() {
    const url = 'http://localhost:11434/api/chat';
    const payload = {
        model: 'llama3.2',
        messages: [{ role: 'user', content: 'Hello, verify connection.' }],
        stream: false
    };

    try {
        console.log(`Connecting to Ollama at ${url}...`);
        const response = await fetch(url, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(payload)
        });

        if (!response.ok) {
            throw new Error(`HTTP Error: ${response.status} - ${response.statusText}`);
        }

        const data = await response.json();
        console.log('✅ Connection Successful!');
        console.log('Response:', data.message.content);
    } catch (error) {
        console.error('❌ Connection Failed:', error.message);
        if (error.cause) console.error('Cause:', error.cause);
        // process.exit(1); 
        // Don't exit with error code yet, just log it. 
        // If it fails, we might need to verify if Ollama is actually running or if model is pulled.
    }
}

checkOllama();
